src/Microsoft.DotNet.Build.CloudTestTasks/CopyBlobsToLatest.cs

```diff
@@ -0,0 +1,231 @@
+
+        [Required]
+        public string Commit { get; set; }
+
+        public bool Coherent { get; set; }
+
+        /// <summary>
+        /// A list of full version strings that should be converted to "latest" for each blob id,
+        /// in addition to ProductVersion.
+        /// </summary>
+        public string[] FullVersions { get; set; }
+
+        /// <summary>
+        /// Forces publish even if the Commit was already published.
+        /// </summary>
+        public bool ForcePublish { get; set; }
+
+        private Regex _versionRegex = new Regex(@"(?<version>\d+\.\d+\.\d+)(-(?<prerelease>[^-]+-)?(?<major>\d+)-(?<minor>\d+))?");
+
+        public override bool Execute()
+        {
+            ParseConnectionString();
+
+            if (Log.HasLoggedErrors)
+            {
+                return false;
+            }
+
+            string sourceDir = $"{Product}/{ProductVersion}/";
+            string channelDir = $"{Product}/{Channel}/";
+            string semaphoreBlob = $"{channelDir}publishSemaphore";
+
+            CreateBlobIfNotExists(semaphoreBlob);
+
+            AzureBlobLease blobLease = new AzureBlobLease(
+                AccountName,
+                AccountKey,
+                ConnectionString,
+                ContainerName,
+                semaphoreBlob,
+                Log);
+
+            Log.LogMessage($"Acquiring lease on semaphore blob '{semaphoreBlob}'");
+            blobLease.Acquire();
+
+            try
+            {
+                string targetVersionFile = $"{channelDir}{Commit}";
+
+                // Prevent race conditions by dropping a version hint of what version this is. If
+                // we see this file and it is the same as our version then we know that a race
+                // happened where two+ builds finished at the same time and someone already took
+                // care of publishing and we have no work to do.
+                if (IsLatestSpecifiedVersion(targetVersionFile) && !ForcePublish)
+                {
+                    Log.LogMessage(
+                        MessageImportance.High,
+                        $"Version '{Commit}' is already published, skipping finalization. " +
+                            $"Hint file: '{targetVersionFile}'");
+
+                    return true;
+                }
+
+                // Delete old version files
+                GetBlobList(channelDir)
+                    .Select(s => s.Replace($"/{ContainerName}/", ""))
+                    .Where(w => _versionRegex.Replace(Path.GetFileName(w), "") == "")
+                    .ToList()
+                    .ForEach(f => TryDeleteBlob(f));
+
+                // Drop the version file signaling such for any race-condition builds (see above comment).
+                CreateBlobIfNotExists(targetVersionFile);
+
+                CopyBlobs(sourceDir, channelDir);
+
+                // Generate the latest version text file
+                string sfxVersion =
```

Comment:
(nit) `sfx` is probably a copy-paste leftover from core-setup.  It stands for "shared framework".

Reply:
Ah, yep, changed to `versionText`.